synapse/handlers/deactivate_account.py

```diff
@@ -255,13 +255,14 @@ async def activate_account(self, user_id: str) -> None:
         Args:
             user_id: ID of user to be re-activated
         """
-        # Add the user to the directory, if necessary.
         user = UserID.from_string(user_id)
-        profile = await self.store.get_profileinfo(user.localpart)
-        await self.user_directory_handler.handle_local_profile_change(user_id, profile)
 
         # Ensure the user is not marked as erased.
         await self.store.mark_user_not_erased(user_id)
 
         # Mark the user as active.
         await self.store.set_user_deactivated_status(user_id, False)
+
+        # Add the user to the directory, if necessary.
```

Comment:
Maybe to help us avoid re-introducing this in the future?

```suggestion
        # Add the user to the directory, if necessary. Note that this must be done
        # after the user is re-activated.
```